### PR TITLE
Add unit tests for Graveler Revert action

### LINKED ISSUE #7445: Add unit tests for revert operation

### PR DESCRIPTION
**Pull request by lynnro314**
Closes #7445

### CODE REVIEW

File: pkg/graveler/graveler_test.go [hunk 1]
```diff
@@ -2292,3 +2292,182 @@ func TestGravelerCreateCommitRecord(t *testing.T) {
 		require.NoError(t, err)
 	})
 }
+
+func TestGraveler_Revert(t *testing.T) {
+	type fields struct {
```
Comment (arielshaqed) [inline pkg/graveler/graveler_test.go:2297]: This name is a bit unpleasant.  Consider "deps"/"dependencies", or even "gravelerDependencies"; you could have "deps" as the name of the type, and "gravelerDeps" as the name of the field?

File: pkg/graveler/graveler_test.go [hunk 2]
```diff
@@ -2292,3 +2292,182 @@ func TestGravelerCreateCommitRecord(t *testing.T) {
...
+		ctx          context.Context
+		branchID     graveler.BranchID
+		ref          graveler.Ref
+		parentNumber int
+		allowEmpty   bool
+	}
+	tests := []struct {
+		name        string
+		fields      fields
+		args        args
```
Comment (arielshaqed) [inline pkg/graveler/graveler_test.go:2311]: This:
```suggestion
		revertArgs        args
```
might make it clear why we have `fields` and `args`.

File: pkg/graveler/graveler_test.go [hunk 3]
```diff
@@ -2292,3 +2292,182 @@ func TestGravelerCreateCommitRecord(t *testing.T) {
...
 	})
 }
+
+func TestGraveler_Revert(t *testing.T) {
+	type fields struct {
+		CommittedManager *testutil.CommittedFake
+		RefManager       *testutil.RefsFake
+	}
+	type args struct {
+		ctx          context.Context
```
Comment (arielshaqed) [inline pkg/graveler/graveler_test.go:2302]: **Go language note.**  This is never used, and probably never should be.  Having it here means encourages passing a `nil` Context, which you should actually never do!  The [`context` godoc](https://pkg.go.dev/context) says:
> Do not pass a nil [Context](https://pkg.go.dev/context#Context), even if a function permits it. Pass [context.TODO](https://pkg.go.dev/context#TODO) if you are unsure about which Context to use.

My recommendation: remove this field, create `ctx := context.TODO()` at the top of the test, and just pass that `ctx` to Revert.

File: pkg/graveler/graveler_test.go (hunk 2 above)
Comment (arielshaqed) [inline pkg/graveler/graveler_test.go:2311]: I think the test cases might be easier to read if we put `args` on the first line and `fields` on the second line.  That way when I read a test case, I know what to expect for each branch or commit name.

As an alternative, you could give some of the branches or commits a longer name -- I give one example below.

File: pkg/graveler/graveler_test.go [hunk 4]
```diff
@@ -2292,3 +2292,182 @@ func TestGravelerCreateCommitRecord(t *testing.T) {
...
+					},
+					Refs: map[graveler.Ref]*graveler.ResolvedRef{
+						"b1": {
+							Type:                   graveler.ReferenceTypeBranch,
+							ResolvedBranchModifier: 0,
+							BranchRecord: graveler.BranchRecord{
+								BranchID: "b1",
+								Branch: &graveler.Branch{
+									CommitID:     "c1",
+									StagingToken: "token",
```
Comment (arielshaqed) [inline pkg/graveler/graveler_test.go:2374]: Followup on comment above about changing object names in the test cases: Here I might prefer
```suggestion
							BranchRecord: graveler.BranchRecord{
								BranchID: "dirty-branch",
								Branch: &graveler.Branch{
									CommitID:     "c1",
									StagingToken: "token",
```

File: pkg/graveler/graveler_test.go [hunk 5]
```diff
@@ -2292,3 +2292,182 @@ func TestGravelerCreateCommitRecord(t *testing.T) {
...
+		allowEmpty   bool
+	}
+	tests := []struct {
+		name        string
+		fields      fields
+		args        args
+		expectedVal graveler.CommitID
+		expectedErr error
+	}{
+		{
```
Comment (arielshaqed) [inline pkg/graveler/graveler_test.go:2315]: Can you add a test case with `allowEmpty: true`?

**Review by arielshaqed — APPROVED** [review]
## Neat!

Thanks.  Sorry for time taken, I was not feeling well on Sunday.

Really well-written clear test, easy to review 180 lines like this.  All my comments are obvious recommendations.  The one that I'd most like to see is a pair of tests for a state that needs `AllowEmpty` to succeed; then we can test that feature too!

**Comment by itaiad200** [comment]
@arielshaqed I addressed your comments trying to wrap up this PR. I'll merge it after tests pass.

**Pull request merged**